pkg/apis/cloudcredential/v1/credentialsrequest_types.go

```diff
@@ -44,17 +44,55 @@ const (
 	CloudCredOperatorConfigMap = "cloud-credential-operator-config"
 )
 
-// NOTE: Run "make" to regenerate code after modifying this file
+// NOTE: Run "make update" to regenerate code after modifying this file
 
 // CredentialsRequestSpec defines the desired state of CredentialsRequest
 type CredentialsRequestSpec struct {
 	// SecretRef points to the secret where the credentials should be stored once generated.
-	SecretRef corev1.ObjectReference `json:"secretRef"`
+	// +optional
+	SecretRef corev1.ObjectReference `json:"secretRef,omitempty"`
+
+	// StorageDestination specifies the type and location of a resource to which the credentials
+	// information can be stored or linked
+	StorageDestination StorageDestination `json:"storage,omitempty"`
 
 	// ProviderSpec contains the cloud provider specific credentials specification.
 	ProviderSpec *runtime.RawExtension `json:"providerSpec,omitempty"`
 }
 
+const (
+	SecretStorageDestinationType         = "Secret"
+	ServiceAccountStorageDestinationType = "ServiceAccount"
+)
+
+// StorageDestination defines the type and location of a resource to which the credentials
+// information can be stored or linked
+// +union
+type StorageDestination struct {
+	// StorageType contains the type of storage that should be used for the credentials
+	// +unionDiscriminator
```

Comment:
Whoah, what does this do?

Reply:
https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/20190325-unions.md

However, afaict, it doesn't have an impact on the OpenAPIv3 schema that is output by the generator :confused: I figured I'd keep it in there though.

https://github.com/openshift/api/search?q=unionDiscriminator

Reply:
Yes, it's best to list this